### PR TITLE
[Az.DataProtection] OnBoard New RP DataProtection To Powershell. PR to Generation branch

### PR DESCRIPTION
**Pull request by sambitratha**
## Description

Adding Az.DataProtection module to powershell. This module has autorest generated cmdlets. Adding Custom folder to generation branch.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+GetBackupFrequenceFromTimeInterval [-RepeatingTimeInterval] <String[]> [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
+```powershell
```
Comment (dolauli) [inline src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md:25]: Please populate all examples like these.

File: src/DataProtection/readme.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+This file contains the configuration for generating My API from the OpenAPI specification.
+
+> see https://aka.ms/autorest
+
+``` yaml
+# it's the same options as command line options, just drop the double-dash!
+require:
+  - $(this-folder)/../readme.azure.noprofile.md
+input-file: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/dataprotection/resource-manager/Microsoft.DataProtection/preview/2021-02-01-preview/dataprotection.json
```
Comment (dolauli) [inline src/DataProtection/readme.md:39]: Replace the master with the commit id to pin the swagger version

File: src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md (hunk 1 above)
Comment (sambitratha) [reply src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md:25]: Hi @dolauli  these are helper scripts which will not be exposed to customers. We use these inside our customized cmdlets and these are not part of Az.DataProtection.psd1. With that in mind, do we still need to give examples for these?

File: src/DataProtection/readme.md (hunk 2 above)
Comment (sambitratha) [reply src/DataProtection/readme.md:39]: okay. will do

File: src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md (hunk 1 above)
Comment (sambitratha) [reply src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md:25]: This is for more information. We've distinguised these helper scripts from actual cmdlets by not  having "-" as a separator between Verb and Subject

Comment (dolauli) [reply src/DataProtection/docs/GetBackupFrequenceFromTimeInterval.md:25]: Add `[Microsoft.Azure.PowerShell.Cmdlets.DataProtection.DoNotExportAttribute()]` as below for all the help functions
```
function AddFilterToQuery {
	[Microsoft.Azure.PowerShell.Cmdlets.DataProtection.DoNotExportAttribute()]
```

**Pull request merged**